docs/proposals/resource-quota.md

```diff
@@ -0,0 +1,404 @@
+  // fieldPath is the path that the selector applies to.
+  FieldPath string `json:"fieldPath" patchStrategy:"merge" patchMergeKey:"fieldPath"`
+  // operator represents a fieldPath's relationship to a set of values.
+  // Valid operators ard In, NotIn, Exists and DoesNotExist.
+  Operator FieldSelectorOperator `json:"operator"`
+  // values is an array of string values. If the operator is In or NotIn,
+  // the values array must be non-empty. If the operator is Exists or DoesNotExist,
+  // the values array must be empty. This array is replaced during a strategic
+  // merge patch.
+  Values []string `json:"values,omitempty"`
+}
+
+// A field selector operator is the set of operators that can be used in a selector requirement.
+type FieldSelectorOperator string
+
+const (
+  FieldSelectorOpIn           FieldSelectorOperator = "In"
+  FieldSelectorOpNotIn        FieldSelectorOperator = "NotIn"
+  FieldSelectorOpExists       FieldSelectorOperator = "Exists"
+  FieldSelectorOpDoesNotExist FieldSelectorOperator = "DoesNotExist"
+)
+```
+
+### ResourceQuota
+
+The `ResourceQuotaSpec` will be modified to support the optional
+selectors to filter the `ResourceQuota` document.
+
+```
+// ResourceQuotaSpec defines the desired hard limits to enforce for Quota
+type ResourceQuotaSpec struct {
+  // Hard is the set of desired hard limits for each named resource
+  Hard ResourceList `json:"hard,omitempty"`
+  // LabelSelector is a label query that must match over each resource that is tracked by quota.
+  // If empty, the quota matches every resource.
+  // More info: http://releases.k8s.io/HEAD/docs/user-guide/labels.md#label-selectors
+  LabelSelector *LabelSelector `json:"labelSelector,omitempty"`
+  // FieldSelector is a field query that must match over the resource tracked by quota.
+  // If empty, the quota matches every resource.
+  FieldSelector *FieldSelector `json:"fieldSelector,omitempty"`
+}
+```
+
+The selectors are immutable post-creation.
+
+API definition of `ReplenishmentPolicy` to support replenishment
+via a rate are deferred pending further prototyping to understand
+what additional fields may or may not be needed to track when
+quota was last replenished.
+
+### Pod
+
+The administrator is able to filter terminating vs non-terminating workloads
+by defining a field selector of `spec.restartPolicy`.
+
+The administrator is not able to filter pods by quality of service.  To
+facilitate this ability, the `Pod` should expose a new field on it's `PodSpec`
+that corresponds to its `QualityOfService` that is set by the server at
+creation and used by `ResourceQuota`.  The administrator would then
+be able to filter quota by tracking `spec.qualityOfService` to quota
+`BestEffort` pods differently than `Burstable` and `Guaranteed`.
+
+## Implementation Phases and Anticipated Issues
+
+Phase 1:
+
+* Support ability for cluster-admin to quota based on workload duration
+ * Add `spec.restartAlways` as a field selector for `Pod`
```

Comment:
Yes,  typo. Meant spec.restartPolicy

On Monday, January 18, 2016, Ben Parees <notifications@github.com> wrote:

> In docs/proposals/resource-quota.md
> <https://github.com/kubernetes/kubernetes/pull/19761#discussion_r50053875>
> :
>
> > +The administrator is able to filter terminating vs non-terminating workloads
> > +by defining a field selector of `spec.restartPolicy`.
> > +
> > +The administrator is not able to filter pods by quality of service.  To
> > +facilitate this ability, the `Pod` should expose a new field on it's `PodSpec`
> > +that corresponds to its `QualityOfService` that is set by the server at
> > +creation and used by `ResourceQuota`.  The administrator would then
> > +be able to filter quota by tracking `spec.qualityOfService` to quota
> > +`BestEffort` pods differently than `Burstable` and `Guaranteed`.
> > +
> > +## Implementation Phases and Anticipated Issues
> > +
> > +Phase 1:
> > +
> > +* Support ability for cluster-admin to quota based on workload duration
> > + * Add `spec.restartAlways` as a field selector for `Pod`
>
> I think you mean spec.restartPolicy here, right? otherwise i'm very
> confused because it sounds like everything is getting defaulted to
> restartAlways which doesn't make sense for builds.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/19761/files#r50053875>.
>